### PR TITLE
Allow creation of new chats by search

### PR DESCRIPTION
**Pull request by ema-fox**
also:
 - show icon of board next to other chat participant
 - fix prepending 'you: ' in chat sidebar if the last message is from the user
 - make hover style in chat sidebar darker to be different from the background

### CODE REVIEW

File: src/sb/app/chat/data.cljc [hunk 1]
```diff
@@ -126,13 +114,48 @@
...
+          sch/wrap-id)))
+
 (q/defquery chat
   "Get a chat by chat-id"
   {:prepare [az/with-account-id!]}
   [{:as params :keys [account-id chat-id]}]
   (ensure-participant! account-id (dl/entity chat-id))
   (q/pull chat-fields-full chat-id))
 
+(q/defquery proto-chat
```
Comment (mhuebert) [inline src/sb/app/chat/data.cljc:134]: is this a thing for dev/testing purposes?

**Review by mhuebert — COMMENTED** [review]
Thank you! I am wondering how we can verify behavior. WYDT of
1. some kind of minimal test(s)
2. doing a short screen recording of new functionality added (eg. clicking through the flow in a browser)?

File: src/sb/app/chat/data.cljc (hunk 1 above)
Comment (ema-fox) [reply src/sb/app/chat/data.cljc:134]: No it's actually being used to load the chat participants and entity for a new chat into the client.

**Comment by ema-fox** [comment]
@mhuebert I'm not seeing any existing tests, any preferences on the testing framework for the chat tests?

**Comment by ema-fox** [comment]
https://github.com/sparkboard/sparkboard/assets/121572/e6a4f2d6-8b76-4afe-9a12-447398e872a1

Here's a quick recording of creating a new chat.

**Comment by ema-fox** [comment]
Something broke again. Looks like I have to make this PR a draft again and investigate.

**Comment by mhuebert** [comment]
No test framework set yet. Open to suggestions.

http://matt.is — Berlin, Germany
--

**Comment by ema-fox** [comment]
It should be working now but the whole approach seems to be a bit brittle. Too many parts of the code are concerning themselves with the difference between new and existing chats. Alas I don't see a better factoring at the moment.

Maybe making `:chat/key` more central could work as this is something already known before the chat entity is created.

**Comment by mhuebert** [comment]
I agree with you that it's brittle. Should we try relying more on
 :chat/key and see if that leaves us in a better place, or if it helps us
discover a better approach?

**Comment by ema-fox** [comment]
I think what makes most sense is for me to write a few tests, merge it, let it sit while I work on other features and then I take another stab at it with a broader understanding of the project.

**Comment by ema-fox** [comment]
Are we gonna want group-chats in the future? In that case using `:chat/key` to identify a chat is gonna be annoying because of possible unbounded size but especially because it's not gonna be stable.

**Comment by mhuebert** [comment]
> Are we gonna want group-chats in the future? In that case using `:chat/key` to identify a chat is gonna be annoying because of possible unbounded size but especially because it's not gonna be stable.

I could argue either side of this one. We could draw a distinction between "direct messages" (what this feature currently is), and "group chats" (future thing), and we do whatever is simplest for "direct messages" now, assuming that if we build a "group chat" thing in the future it is going to require a new/different approach.

Or, if it's really just about this one `:chat/key` issue, and we can find a way to keep all our code reasonably clean/simple while using a unique uuid, then - why not. I'd leave it to your judgment while you've got your head wrapped around the code.

**Review by mhuebert — APPROVED** [review]
See comment.. feel free to merge as you see fit

**Pull request merged**